src/w32_regex_traits.cpp

```diff
@@ -42,12 +42,22 @@ namespace std{
 
 namespace boost{ namespace BOOST_REGEX_DETAIL_NS{
 
-#ifdef BOOST_NO_ANSI_APIS
+#if defined(BOOST_NO_ANSI_APIS) || defined (BOOST_ASIO_WINDOWS_RUNTIME) || defined (BOOST_PLAT_WINDOWS_PHONE)
 UINT get_code_page_for_locale_id(lcid_type idx)
 {
    WCHAR code_page_string[7];
+#if defined(BOOST_NO_ANSI_APIS)
```

Comment:
When building for WinRT, BOOST_NO_ANSI_APIS is always defined (see https://github.com/boostorg/config/pull/80), so this won't work. It also makes some of the added checks redundant.